dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -101,39 +104,58 @@ def manual_pass
     redirect_to :manual_pass_form
   end
 
+  # get /admin/permissions
   def permissions_form
+    search_term = params[:search_term]
+    permission = params[:permission]
+    if search_term.present?
+      if search_term =~ /^\d+$/
+        @user = User.find(search_term)
+      else
+        users = User.where(hashed_email: User.hash_email(search_term))
+        @user = users.first
+        if users.count > 1
+          flash[:notice] = "More than one User matches email address.  "\
+                         "Showing first result.  Matching User IDs - #{users.map(&:id).join ','}"
+        end
+      end
+      unless @user || search_term.blank?
+        flash[:notice] = "User Not Found"
+      end
+    elsif permission.present?
+      @users_with_permission = User.joins(:permissions).where(user_permissions: {permission: permission}).order(:email)
```

Comment:
The only fields going to the client are specified in the view [here](https://github.com/sureshc/code-dot-org/blob/2d05a99e5eaca465c0af965ad355e9de63dc1a70/dashboard/app/views/admin_users/permissions_form.html.haml#L61-L67). Are you saying you want to scope the model passed between controller and view to only contain those fields?

As an aside I'm not convinced we need to display all the fields we are displaying in the view, but it is an admin-only page.